File: src-tauri/src/models/import_export.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

// ===== 数据导出相关结构体 =====

/// 数据导出请求
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExportRequest {
    /// 数据来源（查询或表）
    pub source: ExportSource,
    /// 导出格式
    pub format: ExportFormat,
    /// 导出文件路径
    pub file_path: String,
    /// 格式特定选项
    pub options: ExportOptions,
}

/// 导出数据来源
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum ExportSource {
    /// 自定义 SQL 查询
    Query { sql: String, database: String },
    /// 整表导出
    Table { database: String, table: String },
}

/// 导出文件格式
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum ExportFormat {
    Csv,
    Json,
    Sql,
    Excel,
    Markdown,
}

/// 导出格式选项
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExportOptions {
    /// CSV 分隔符（默认逗号）
    pub csv_delimiter: Option<String>,
    /// CSV 文本限定符（默认双引号）
    pub csv_quote_char: Option<String>,
    /// CSV 是否包含列标题（默认 true）
    pub csv_include_header: Option<bool>,
    /// SQL 导出目标表名
    pub sql_table_name: Option<String>,
    /// SQL 导出是否包含 CREATE TABLE 语句
    pub sql_include_create: Option<bool>,
    /// SQL 导出每批 INSERT 的行数（默认 1000）
    pub sql_batch_size: Option<u32>,
    /// 文件编码（默认 UTF-8）
    pub encoding: Option<String>,
}

/// 导出结果
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExportResult {
    /// 是否成功
    pub success: bool,
    /// 导出行数
    pub row_count: u64,
    /// 文件大小（字节）
    pub file_size: u64,
    /// 错误信息
    pub error: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionExport {
    pub version: i32,
    pub exported_at: i64,
    pub connections: Vec<ConnectionExportItem>,
    pub groups: Vec<ConnectionGroupExport>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionExportItem {
    pub name: String,
    #[serde(rename = "type")]
    pub conn_type: String,
    pub group_name: Option<String>,
    pub host: String,
    pub port: i32,
    pub username: String,
    pub password: Option<String>,
    pub config: HashMap<String, serde_json::Value>,
    pub color: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionGroupExport {
    pub name: String,
    pub parent_name: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ImportOptions {
    pub conflict_strategy: ConflictStrategy,
    pub import_passwords: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum ConflictStrategy {
    Skip,
    Overwrite,
    Rename,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ImportResult {
    pub success: bool,
    pub imported: i32,
    pub skipped: i32,
    pub failed: i32,
    pub errors: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ImportPreview {
    pub connections: Vec<ConnectionExportItem>,
    pub groups: Vec<ConnectionGroupExport>,
    pub conflicts: Vec<String>,
}

File: src/api/import-export.ts
import { invokeCommand } from '@/api/base'

// 导出格式
export interface ConnectionExport {
  version: number  // 导出格式版本
  exportedAt: number  // 导出时间戳
  connections: ConnectionExportItem[]
  groups: ConnectionGroupExport[]
}

export interface ConnectionExportItem {
  name: string
  type: string  // 'database' | 'ssh' | 'sftp'
  groupName?: string  // 分组名称（而不是 ID）
  host: string
  port: number
  username: string
  password?: string  // 可选，明文密码
  config: Record<string, any>  // 类型特定配置
  color?: string
}

export interface ConnectionGroupExport {
  name: string
  parentName?: string  // 父分组名称
}

// 导入选项
export interface ImportOptions {
  conflictStrategy: 'skip' | 'overwrite' | 'rename'  // 冲突处理策略
  importPasswords: boolean  // 是否导入密码
}

// 导入结果
export interface ImportResult {
  success: boolean
  imported: number  // 成功导入的连接数
  skipped: number  // 跳过的连接数
  failed: number  // 失败的连接数
  errors: string[]  // 错误信息
}

// 导入预览
export interface ImportPreview {
  connections: ConnectionExportItem[]
  groups: ConnectionGroupExport[]
  conflicts: string[]  // 冲突的连接名称
}

// API 函数

export function exportConnections(connectionIds?: string[]): Promise<ConnectionExport> {
  return invokeCommand('export_connections', { connectionIds })
}

export function importConnections(
  data: ConnectionExport,
  options: ImportOptions
): Promise<ImportResult> {
  return invokeCommand('import_connections', { data, options })
}

export function previewImport(data: ConnectionExport): Promise<ImportPreview> {
  return invokeCommand('preview_import', { data })
}

// Navicat 导入
export function importNavicatXml(xmlContent: string): Promise<ConnectionExport> {
  return invokeCommand('import_navicat_xml', { xmlContent })
}

// Termius 导入
export function importTermiusJson(jsonContent: string): Promise<ConnectionExport> {
  return invokeCommand('import_termius_json', { jsonContent })
}
